tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradeStep.java

```diff
@@ -234,7 +220,7 @@ public void setStatus(UpgradeStepStatus status) {
 
 		_status = status;
 
-		UpgradePlanner upgradePlanner = _serviceTracker.getService();
+		UpgradePlanner upgradePlanner = ServicesLookup.getSingleService(UpgradePlanner.class, null);
```

Comment:
I have no ideas why serviceTracker would not work here, getService will return null. I switch to ServiceLookup(ServiceReference way) to get service

Reply:
WE can switch it to _serviceTracker.open() and then later on when we do _serviceTracker.getService() it will be correct.